### PR TITLE
Feature/Be able to turn off the spanned text for custom notification layout

### PR DESCRIPTION
**Pull request by kmalmur**
Added a flag to the builder whether or not the spanned text should be used for the custom notification layout. For some cases we want the spanned text to be used only for the regular notification style, without propagation to the custom one, as it might define a totally different style of the text.

### CODE REVIEW

File: library/src/main/java/br/com/goncalves/pugnotification/notification/Custom.java [hunk 1]
```diff
@@ -63,21 +73,25 @@ private void setSmallIcon() {
     }
 
     public Custom background(@DrawableRes int resource) {
-        if (resource <= 0) {
-            throw new IllegalArgumentException("Resource ID Should Not Be Less Than Or Equal To Zero!");
+        if (resource == 0) {
```
Comment (Messady) [inline library/src/main/java/br/com/goncalves/pugnotification/notification/Custom.java]: @kmalmur Isn't `-1` considered as a possible value and an invalid ResourceId?

Comment (kmalmur) [reply library/src/main/java/br/com/goncalves/pugnotification/notification/Custom.java]: According to the lint message, it should be checked only against 0 as all the negative values are valid. Also from what I checked the doc doesn't say anything about -1. Did you find it somewhere?

Comment (aminezghidi) [reply library/src/main/java/br/com/goncalves/pugnotification/notification/Custom.java]: I would suggest to keep the non-negative check.
https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/content/res/ResourceId.java#39

File: library/src/main/java/br/com/goncalves/pugnotification/notification/Load.java [hunk 2]
```diff
@@ -22,14 +22,14 @@
...
 public class Load {
-    private static final String TAG = "Pugnotification.Load";
     private NotificationCompat.Builder builder;
     private String title;
     private String message;
     private Spanned messageSpanned;
     private String tag;
     private int notificationId;
     private int smallIcon;
+    private boolean useSpanForCustomNotification;
```
Comment (aminezghidi) [inline library/src/main/java/br/com/goncalves/pugnotification/notification/Load.java]: Shouldn't this be initialized to `true` so we don't accidentally change behaviour of notifications that already use spanned text?

File: library/src/main/java/br/com/goncalves/pugnotification/notification/Custom.java [hunk 3]
```diff
@@ -24,10 +24,19 @@
     private int mBackgroundResId;
     private int mPlaceHolderResourceId;
     private ImageLoader mImageLoader;
-
-
-    public Custom(NotificationCompat.Builder builder, int identifier, String title, String message, Spanned messageSpanned, int smallIcon, String tag) {
+    private final boolean mUseSpanForCustomNotification;
```
Comment (aminezghidi) [inline library/src/main/java/br/com/goncalves/pugnotification/notification/Custom.java]: Can you please remove the `m` prefix from all variables?

**Review by aminezghidi — CHANGES_REQUESTED** [review]

File: library/src/main/java/br/com/goncalves/pugnotification/notification/Custom.java (hunk 1 above)
Comment (kmalmur) [reply library/src/main/java/br/com/goncalves/pugnotification/notification/Custom.java]: Non negative check causes the lint issue, but the snippet you provided seems to do exactly what we want to do. Thanks

File: library/src/main/java/br/com/goncalves/pugnotification/notification/Custom.java (hunk 3 above)
Comment (kmalmur) [reply library/src/main/java/br/com/goncalves/pugnotification/notification/Custom.java]: Ok. I wasn't sure what approach we want to take as this is a fork of the existing library

**Comment by kmalmur** [comment]
@aminezghidi changes implemented

File: library/src/main/java/br/com/goncalves/pugnotification/notification/Load.java [hunk 4]
```diff
@@ -51,15 +52,19 @@ public Load tag(@NonNull String tag) {
     }
 
     public Load title(@StringRes int title) {
-        if (title <= 0) {
-            throw new IllegalArgumentException("Resource ID Should Not Be Less Than Or Equal To Zero!");
+        if (title == 0) {
```
Comment (aminezghidi) [inline library/src/main/java/br/com/goncalves/pugnotification/notification/Load.java]: Can we use `ResourceUtils.assertResouceValid(resource);` here?

File: library/src/main/java/br/com/goncalves/pugnotification/notification/Load.java [hunk 5]
```diff
@@ -75,8 +80,8 @@ public Load title(String title) {
     }
 
     public Load message(@StringRes int message) {
-        if (message <= 0) {
-            throw new IllegalArgumentException("Resource ID Should Not Be Less Than Or Equal To Zero!");
+        if (message == 0) {
```
Comment (aminezghidi) [inline library/src/main/java/br/com/goncalves/pugnotification/notification/Load.java]: same here

File: library/src/main/java/br/com/goncalves/pugnotification/notification/Load.java [hunk 6]
```diff
@@ -105,8 +110,8 @@ public Load message(@NonNull Spanned messageSpanned) {
     }
 
     public Load color(@ColorRes int color) {
-        if (color <= 0) {
-            throw new IllegalArgumentException("Resource ID Should Not Be Less Than Or Equal To Zero!");
+        if (color == 0) {
```
Comment (aminezghidi) [inline library/src/main/java/br/com/goncalves/pugnotification/notification/Load.java]: same here

File: library/src/main/java/br/com/goncalves/pugnotification/notification/Load.java [hunk 7]
```diff
@@ -119,8 +124,8 @@ public Load color(@ColorRes int color) {
     }
 
     public Load ticker(@StringRes int ticker) {
-        if (ticker <= 0) {
-            throw new IllegalArgumentException("Resource ID Should Not Be Less Than Or Equal To Zero!");
+        if (ticker == 0) {
```
Comment (aminezghidi) [inline library/src/main/java/br/com/goncalves/pugnotification/notification/Load.java]: same here

File: library/src/main/java/br/com/goncalves/pugnotification/notification/Load.java [hunk 8]
```diff
@@ -142,25 +147,25 @@ public Load ticker(String ticker) {
 
     public Load when(long when) {
         if (when <= 0) {
```
Comment (aminezghidi) [inline library/src/main/java/br/com/goncalves/pugnotification/notification/Load.java:137]: same here

File: library/src/main/java/br/com/goncalves/pugnotification/notification/Load.java [hunk 9]
```diff
@@ -142,25 +147,25 @@ public Load ticker(String ticker) {
...
         }
 
         this.builder.setWhen(when);
         return this;
     }
 
     public Load bigTextStyle(@StringRes int bigTextStyle) {
-        if (bigTextStyle <= 0) {
-            throw new IllegalArgumentException("Resource ID Should Not Be Less Than Or Equal To Zero!");
+        if (bigTextStyle == 0) {
```
Comment (aminezghidi) [inline library/src/main/java/br/com/goncalves/pugnotification/notification/Load.java]: same here

File: library/src/main/java/br/com/goncalves/pugnotification/notification/Load.java [hunk 10]
```diff
@@ -142,25 +147,25 @@ public Load ticker(String ticker) {
...
         }
 
         return bigTextStyle(PugNotification.singleton.context.getResources().getString(
                 bigTextStyle), null);
     }
 
     public Load bigTextStyle(@StringRes int bigTextStyle, @StringRes int summaryText) {
-        if (bigTextStyle <= 0) {
-            throw new IllegalArgumentException("Resource ID Should Not Be Less Than Or Equal To Zero!");
+        if (bigTextStyle == 0) {
```
Comment (aminezghidi) [inline library/src/main/java/br/com/goncalves/pugnotification/notification/Load.java]: same here

File: library/src/main/java/br/com/goncalves/pugnotification/notification/Load.java [hunk 11]
```diff
@@ -270,8 +280,8 @@ public Load largeIcon(@NonNull Bitmap bitmap) {
     }
 
     public Load largeIcon(@DrawableRes int largeIcon) {
-        if (largeIcon <= 0) {
-            throw new IllegalArgumentException("Resource ID Should Not Be Less Than Or Equal To Zero!");
+        if (largeIcon == 0) {
```
Comment (aminezghidi) [inline library/src/main/java/br/com/goncalves/pugnotification/notification/Load.java]: same here

File: library/src/main/java/br/com/goncalves/pugnotification/notification/Load.java [hunk 12]
```diff
@@ -341,8 +351,8 @@ public Load addPerson(@NonNull String uri) {
     }
 
     public Load button(@DrawableRes int icon, @StringRes int title, @NonNull PendingIntent pendingIntent) {
-        if (title <= 0) {
-            throw new IllegalArgumentException("Resource ID Should Not Be Less Than Or Equal To Zero!");
+        if (title == 0) {
```
Comment (aminezghidi) [inline library/src/main/java/br/com/goncalves/pugnotification/notification/Load.java]: same here

File: library/src/main/java/br/com/goncalves/pugnotification/notification/Load.java [hunk 13]
```diff
@@ -354,8 +364,8 @@ public Load button(@DrawableRes int icon, @NonNull String title, @NonNull Pendin
     }
 
     public Load button(@DrawableRes int icon, @StringRes int title, @NonNull PendingIntentNotification pendingIntentNotification) {
-        if (title <= 0) {
-            throw new IllegalArgumentException("Resource ID Should Not Be Less Than Or Equal To Zero!");
+        if (title == 0) {
```
Comment (aminezghidi) [inline library/src/main/java/br/com/goncalves/pugnotification/notification/Load.java]: same here

File: library/src/main/java/br/com/goncalves/pugnotification/notification/Load.java [hunk 14]
```diff
@@ -51,15 +52,19 @@ public Load tag(@NonNull String tag) {
...
+            return throwResourceCannotBeZeroException();
         }
 
         this.title = PugNotification.singleton.context.getResources().getString(title);
         this.builder.setContentTitle(this.title);
         return this;
     }
 
+    private Load throwResourceCannotBeZeroException() {
```
Comment (aminezghidi) [inline library/src/main/java/br/com/goncalves/pugnotification/notification/Load.java]: This method can be removed now.

File: README.md [hunk 15]
```diff
@@ -35,7 +35,7 @@ PugNotification.with(context)
     .title(title)
     .message(message)
     .bigTextStyle(bigtext)
-    .smallIcon(smallIcon)
+    .smallIconId(smallIconId)
```
Comment (aminezghidi) [inline README.md]: The method has not been renamed.

**Review by aminezghidi — CHANGES_REQUESTED** [review]
@kmalmur few things to change please.

**Comment by kmalmur** [comment]
@aminezghidi changes implemented, thanks for the feedback

**Review by aminezghidi — APPROVED** [review]
Looks good to me @kmalmur 👍 I will let you merge this.

**Pull request merged**